content/en/blog/_posts/xxxx-xx-xx-Fine-grained-SupplementalGroups-control/index.md

```diff
@@ -0,0 +1,160 @@
+---
+layout: blog
+title: 'Kubernetes 1.31: Fine-grained SupplementalGroups control'
+date: xxxx-xx-xx
+slug: fine-grained-supplementalgroups-control
+author: >
+  Shingo Omura (Woven By Toyota)
+
+---
+
+This blog discusses a new feature in Kubernetes 1.31 to improve the handling of supplementary groups in containers in Pods.
+
+
+## Motivation: Implicit group memberships defined in `/etc/group` in the container image
+
+Although this behavior may not be quite popular by many Kubernetes cluster users/admins, kubernetes, by default, _merges_ group information from the Pod with information defined in `/etc/group` in the container image.
+
+Let's see an example, below Pod specifies `runAsUser=1000`, `runAsGroup=3000` and `supplementalGroups=4000` in the Pod's security context.
+
+{{% code_sample file="implicit-groups.yaml" %}}
+
+What is the result of `id` command in the `ctr` container?
+
+```shell
+# Create the Pod:
+$ kubectl apply -f https://k8s.io/blog/xxxx-xx-xx-Fine-grained-SupplementalGroups-control/implicit-groups.yaml
+
+# Verify that the Pod's Container is running:
+$ kubectl get pod implicit-groups
+
+# Check the id command
+$ kubectl exec implicit-groups -- id
+```
+
+Then, output should be similar to this:
+
+```none
+uid=1000 gid=3000 groups=3000,4000,50000
+```
+
+Where does group ID `50000` in supplementary groups (`groups` field) come from even though `50000` is not defined in the Pod's manifest at all? The answer is `/etc/group` file in the container image.
+
+Checking the contents of `/etc/group` in the container image should show below:
+
+```shell
+$ kubectl exec implicit-groups -- cat /etc/group
+...
+user-defined-in-image:x:1000:
+group-defined-in-image:x:50000:user-defined-in-image
+```
+
+Aha! The container's primary user `1000` belongs to the group `50000` in the last entry.
+
+Thus, the group membership defined in `/etc/group` in the container image for the container's primary user is _implicitly_ merged to the information from the Pod.
+
+### What's wrong with it?
+
+The _implicitly_ merged group information from `/etc/group` in the container image may cause some concerns particularly in accessing volumes (see [kubernetes/kubernetes#112879](https://issue.k8s.io/112879) for details) because file permission is controlled by uid/gid in Linux. Even worse, the implicit gids from `/etc/group` can not be detected/validated by any policy engines becuase there is no clue for the implicit group information in the manifest. This can also be concern for Kubernetes security.
```

Comment:
because*

Reply:
Sorry, I don't get it. Could you kindly explain a bit more??

Reply:
you spelled because becuase once :)

Reply:
Oh....　I found it finally😅 Thanks 🙇 🙇 

```suggestion
The _implicitly_ merged group information from `/etc/group` in the container image may cause some concerns particularly in accessing volumes (see [kubernetes/kubernetes#112879](https://issue.k8s.io/112879) for details) because file permission is controlled by uid/gid in Linux. Even worse, the implicit gids from `/etc/group` can not be detected/validated by any policy engines because there is no clue for the implicit group information in the manifest. This can also be a concern for Kubernetes security.
```